pkg/testcase/clusterreset.go

```diff
@@ -0,0 +1,98 @@
+package testcase
+
+import (
+	"fmt"
+
+	"github.com/rancher/distros-test-framework/factory"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestClusterReset() {
+	cluster := factory.ClusterConfig(GinkgoT())
+
+	killall(cluster)
+	shared.LogLevel("INFO", "%s-service killed", cluster.Config.Product)
+
+	stopServer(cluster)
+	shared.LogLevel("INFO", "%s-service stopped", cluster.Config.Product)
+
+	productLocationCmd := fmt.Sprintf("which %s", cluster.Config.Product)
+	productLocation, productLocationErr := shared.RunCommandOnNode(productLocationCmd, cluster.ServerIPs[0])
+	Expect(productLocationErr).NotTo(HaveOccurred())
+	resetCmd := fmt.Sprintf("sudo %s server --cluster-reset", productLocation)
+	shared.LogLevel("INFO", "running cluster reset on server %s\n", cluster.ServerIPs[0])
+
+	if cluster.Config.Product == "k3s" {
+		// k3s cluster reset output returns stdout channel
+		resetRes, resetCmdErr := shared.RunCommandOnNode(resetCmd, cluster.ServerIPs[0])
+		Expect(resetCmdErr).NotTo(HaveOccurred())
+		Expect(resetRes).To(ContainSubstring("Managed etcd cluster"))
+		Expect(resetRes).To(ContainSubstring("has been reset"))
+	} else if cluster.Config.Product == "rke2" {
+		// rke2 cluster reset output returns stderr channel
+		_, resetCmdErr := shared.RunCommandOnNode(resetCmd, cluster.ServerIPs[0])
+		Expect(resetCmdErr).To(HaveOccurred())
+		Expect(resetCmdErr.Error()).To(ContainSubstring("Managed etcd cluster"))
+		Expect(resetCmdErr.Error()).To(ContainSubstring("has been reset"))
+	}
+	shared.LogLevel("INFO", "cluster reset successful")
+
+	deleteDataDirectories(cluster)
+	shared.LogLevel("INFO", "data directories deleted")
+	startServer(cluster)
+	shared.LogLevel("INFO", "%s-service started", cluster.Config.Product)
+}
+
+func killall(cluster *factory.Cluster) {
+	for i := len(cluster.ServerIPs) - 1; i > 0; i-- {
+		productLocationCmd := fmt.Sprintf("which %s", cluster.Config.Product)
```

Comment:
Let's also change this to the same as above, with find:
```suggestion
		productLocationCmd := fmt.Sprintf("sudo find / -type f -executable -name %s", cluster.Config.Product)
```